tasks/cnm-to-cma/cnm_to_cma.py

```diff
@@ -0,0 +1,189 @@
+"""lambda function used to translate CNM messages to CMA messages in aws lambda with cumulus"""
+
+import os
+import re
+from datetime import datetime, timezone
+from typing import List
+from cumulus_logger import CumulusLogger
+from cumulus_process import Process
+
+REGION = os.environ.get("REGION", "us-west-2")
+cumulus_logger = CumulusLogger("cnm2cma")
+
+
+class CNM2CMA(Process):
+    """
+    Message transformation class to transform CNM to CMA
+
+    Attributes
+    ----------
+    logger: logger
+        cumulus logger
+    config: dictionary
+        configuration from cumulus
+    """
+
+    def __init__(self, *args, **kwargs):
+        """class init function"""
+
+        # super().__init__(*args, **kwargs)
+        super(CNM2CMA, self).__init__(*args, **kwargs)
+        self.logger = cumulus_logger
+
+    def process(self):
+        """Main process to transform cnm to cma
+
+        Return
+        ----------
+        dict
+            Payload that is returned as the cma which is a dictionary with list of granules
+        """
+        # This is the payload
+        cnm = self.input
+        # This is the config
+        config = self.config
+        self.logger.info(f"cnm2cma object: {cnm}")
+        self.logger.info(f"config object: {config}")
+
+        granule_id_extraction = config.get("collection").get("granuleIdExtraction")
+        granule_id = cnm.get("product").get("name", "")
+        self.logger.info(f"extracted granule_id: {granule_id}")
+        if "/" in granule_id:
+            granule_id = granule_id[granule_id.index("/") + 1 :]
+        matcher = re.search(granule_id_extraction, granule_id)
+        if matcher:
+            granule_id = matcher.group(1)
+        product = cnm["product"]
+        # extract files or filegroups from product
+        input_files = self.get_input_files(product)
+        files = self.get_files(input_files)
+        self.logger.info(f"files: {files}")
+
+        granule = {
+            "granuleId": granule_id,
+            "version": config.get("collection", {}).get("version"),
+            "dataType": config.get("collection", {}).get("name"),
+            "files": files,
+        }
+        granule_array = [granule]
+        now_as_iso = (
+            datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
+        )
+        cnm["receivedTime"] = now_as_iso
+
+        output = {"cnm": cnm, "output_granules": {"granules": granule_array}}
+        return output
+
+    def get_files(self, input_files: List) -> List:
```

Comment:
May want to consider stronger typing. Currently get files says take any list and return any list. So a list of numbers could be passed in and a list of letters passed out which is not what I think we want when we scan with mypy.